impress-org/givewp
# Update warning message for unsuccessful PayPal account onboarding

Resolves #5517 #5518 

## Description

We do not have a step to reproduce this step. We need more information to catch the cause of this issue. As per the initial finding, it seems that the cause of the error is on the PayPal server which we can not control. We are actively working with the PayPal team to resolve this issue. I added merchant status check API response in the warning message review list. I also added an action step to resolve the on-boarding issue. 



## Affects

I refactor code to take benefit of DI which does not break the existing features.

## Visuals

Error message sample for invalid merchant id:
![image](https://user-images.githubusercontent.com/1784821/102648834-3842de80-418e-11eb-9234-39a7fb41fbd9.png)



## Pre-review Checklist




## Testing Instructions

- Except `merchant status` check all PayPal error consider as unknow that means error message will be different (a little bit ).
- PayPal error message display on basis of country region ( North America vs all ).  An error message will modify on the basis of the PayPal account country.

![image](https://user-images.githubusercontent.com/1784821/102783945-b1764780-43c1-11eb-9514-c0fbae04d047.png)

- **Notice on PayPal Modal close:**
A notice will display when the admin closes the PayPal modal without completing onboarding steps.


- **Notice on missing merchant id:**
You will get notice an `$errors array in `onBoardingRedirectHandler::savePayPalMerchantDetails` under `if` condition. set `if` condition to `true ` to force this error.

- **Notice on invalid client id and secret:**
You will get notice an `$errors array in `onBoardingRedirectHandler::didWeGetValidSellerRestApiCredentials` under `if` condition. set `if` condition to `true ` to force this error.

- **Notice on failed to create webhook:**
You will get notice an `$errors array in `onBoardingRedirectHandler::setUpWebhook` under `catch` condition. Throw exception from `try` to execute  code in `catch`.

- **Notice on failing merchant status check:**
To generate error you can unset `primary_email_confirmed` array key from `$onBoardedData` array in `onBoardingRedirectHandler::isAdminSuccessfullyOnBoarded` function. You must see updated notice as shown in above screenshots.

- **Hide PayPal gateway option on frontend**
Admin will be connected successfully even he failed on `merchant status check` that means PayPal gateway is not ready to accept payment, so it will be from the frontend. To generate error you can unset `primary_email_confirmed` array key from `$onBoardedData` array in `onBoardingRedirectHandler::isAdminSuccessfullyOnBoarded` function. You must see updated notice as shown in the above screenshots.
